internal/net/net.go

```diff
@@ -69,18 +79,22 @@ func Parse(addr string) (host string, port uint16, isIP bool, err error) {
 	return host, port, isIP, err
 }
 
+// IsIPv6 returns weather the address is IPv6 address.
 func IsIPv6(addr string) bool {
 	return net.ParseIP(addr) != nil && strings.Count(addr, ":") >= 2
 }
 
+// IsIPv4 returns weather the address is IPv4 address.
 func IsIPv4(addr string) bool {
 	return net.ParseIP(addr) != nil && strings.Count(addr, ":") < 2
 }
 
+// SplitHostPort splits the address, and return the host/IP address and the port number,
+// and any error occurred.
 func SplitHostPort(hostport string) (host string, port uint16, err error) {
 	switch {
 	case strings.HasPrefix(hostport, "::"):
-		hostport = localIPv6 + hostport
+		hostport = localIPv6 + hostport[2:]
```

Comment:
The format of IPv6 address with port number should be `[addr]:port`, the logic of line 96 is incorrect, should I fix it? @kpango @rinx

Reply:
or if the `hostport` do not contains port number, there are no `[]` to wrap the address, instead the host should use the `hostport` with port `defaultPort`

Reply:
I suggest:
```go
switch {
case hostport == "::1":
    hostport = "[" + localIPv6 + "]:" + defaultPort
........
```

Reply:
or we can return directly.
```go
switch {
case hostport == "::1":
    return localIPv6, defaultPort, nil

Reply:
I think we have to make 2 changes, this case was implemented for detecting loopback address, if there is no port definition on hostport string, we can use default port but in this case we have to check there is a port in hostport string or not, and if exists we have to parse it.
one more thing as you mentioned, loopback addr is `::1` but it also can be written as `::` or `0:0:0:0:0:0:0:1`. so we have to check them

Reply:
@kpango 
I think this switch statement is checking if we can use the default port for `hostport`.  
For the case where the `hostport` string contains port number, I think we can rely on `net.SplitHostPort()` in line 101.

For the case without port number, I think we need to check if it is a valid IPv6 address, and return the default port number.

So I think it is correct, what do you think?
```go
switch{
case !strings.HasPrefix(hostport, "[") && IsIPv6(hostport):
     return hostport, defaultPort, nil
.......
```

Reply:
could you please read RFC more carefuly. In RFC5952 (https://tools.ietf.org/html/rfc5952#section-6) there is 6 types of ipv6 w/ port format we should support them.
you code seems support only 1 format

Reply:
@kpango 
May I confirm that this switch case statement is checking if `hostport` contains hostname or port number only and use the default value if not exists? 

For example, line 98-99 use the default local IPv4 address if the host is missing.
I think IPv6 do not have `empty host` case so we don't need to support it.

But I think IPv6 have `empty port`, so I wrote the logic above.

BTW `net.SplitHostPort` doesn't support all RFC format.
https://golang.org/src/net/ipsock.go?s=4766:4832#L146

Reply:
how should we work on this?

Reply:
```suggestion
		// hostport = localIPv6 + hostport[2:]
```

Reply:
https://github.com/vdaas/vald/projects/3#card-43504189